src/modules/mount/main.py

```diff
@@ -61,6 +61,97 @@ def get_btrfs_subvolumes(partitions):
     return btrfs_subvolumes
 
 
+def parse_global_storage(gs_value):
+    """ Something in the chain is converting on and off to true and false.  This converts it back.
+
+    :param gs_value: The value from global storage which needs to be fixed
+    :return:
+    """
+    if gs_value is True:
+        return "on"
+    elif gs_value is False:
+        return "off"
+    else:
+        return gs_value
+
+
+def mount_zfs(root_mount_point, partition):
+    """ Mounts a zfs partition at @p root_mount_point
+
+    :param root_mount_point: The absolute path to the root of the install
+    :param partition: The partition map from global storage for this partition
+    :return:
+    """
+    # Get the list of zpools from global storage
+    zfs_pool_list = libcalamares.globalstorage.value("zfsPoolInfo")
+    if not zfs_pool_list:
+        libcalamares.utils.warning("Failed to locate zfsPoolInfo data in global storage")
+        raise Exception("Internal error mounting zfs datasets")
+
+    # Find the zpool matching this partition
+    for zfs_pool in zfs_pool_list:
+        if zfs_pool["mountpoint"] == partition["mountPoint"]:
+            pool_name = zfs_pool["poolName"]
+            ds_name = zfs_pool["dsName"]
+
+    # import the zpool
+    import_result = subprocess.run(['zpool', 'import', '-R', root_mount_point, pool_name])
+    if import_result.returncode != 0:
+        raise Exception("Failed to import zpool")
```

Comment:
These exceptions seem rather .. generic exceptions. Is there a more specific type you can use here? If these exceptions can escape from the module, it's going to create an ugly failure message. If you can catch the specific type and turn the exception into a failure return from the module -- with a human-readable message -- then it becomes much nicer for the user. ("nicer" is relative: the installation still failed then :( )

Reply:
In that case, I think I am doing the wrong thing here.  I am raising those exceptions to provide a human readable error to the user.

If the proper way to do that is for the module to return an error string of some sort, I will look into making it do that instead.